### PR TITLE
[Bug fix] LPS-130946

### PR DESCRIPTION
**Pull request by dantewang**
@tinatian This is caused by jodd upgrade. The DateConverter in the new jodd-util assumes the date string to be either of ISO_DATE_TIME format or a linux timestamp. But we are sending yyyy-MM-dd or yyyy-M-d (in Poshi). I added logic to build a custom `DateTimeFormatter` to handle this instead of copying the old code from old jodd.

- LPS-130946 DateConverter in jodd 6 uses Java 8 LocalDateTime.parse to parse the date string, which sticks to a strict pattern. Provide a converter to use DateTimeFormatter with optional parts to support multiple formats in string parsing.
- LPS-130946 Old jodd also supports yyyy-M-d as Poshi is using this format

### CODE REVIEW

**Comment by dantewang** [comment]
@tinatian Previous passed ci on https://github.com/Tim-Cao/liferay-portal/pull/624 and also manually checked by @Tim-Cao.

**Comment by dantewang** [comment]
ci:test:sf

**Comment by dantewang** [comment]
ci:test:relevant

File: portal-impl/src/com/liferay/portal/jsonwebservice/JSONWebServiceActionImpl.java [hunk 1]
```diff
@@ -682,9 +682,13 @@ public Date convert(Object object) {
 
 		private static final DateTimeFormatter _dateTimeFormatter =
 			new DateTimeFormatterBuilder().parseCaseInsensitive(
-			).append(
+			).appendOptional(
 				DateTimeFormatter.ISO_LOCAL_DATE
 			).optionalStart(
+			).appendPattern(
+				"yyyy-M-d"
```
Comment (tinatian) [inline portal-impl/src/com/liferay/portal/jsonwebservice/JSONWebServiceActionImpl.java:689]: @dantewang Just want to confirm is "yyyy-M-d" the only one missing?

Tina.

Comment (dantewang) [reply portal-impl/src/com/liferay/portal/jsonwebservice/JSONWebServiceActionImpl.java:689]: @tinatian yyyy-MM-dd + yyyy-M-d should cover what the old jodd supports for dates.

Comment (tinatian) [reply portal-impl/src/com/liferay/portal/jsonwebservice/JSONWebServiceActionImpl.java:689]: @dantewang But here only "yyyy-M-d" is set, what about the other one?

Comment (dantewang) [reply portal-impl/src/com/liferay/portal/jsonwebservice/JSONWebServiceActionImpl.java:689]: @tinatian `ISO_LOCAL_DATE` is `yyyy-MM-dd`.

**Comment by dantewang** [comment]
See https://github.com/tinatian/liferay-portal/pull/2240

**Pull request closed without merging**